### DISCUSSION TITLE
Aspect Paint

### DISCUSSION THREAD

**Discussion by redstone-dev** [AspectLinux/aspectlinux.github.io#9] [Ideas]
Basically a paint app, but it works in a TUI.

**Comment by HGT-123** [comment]
Whats a TUI?

**Reply by redstone-dev** [reply]
**T**ext **U**ser **I**nterface. Basically the terminal.

**Comment by redstone-dev** [comment]
@michaeleldar I'm going to assign you to this project

**Comment by Nonethousand** [comment]
I actually have just the thing for this.
A little bit ago I made a tool that can detect the mouse position relative to the terminal, and it returns the line and column the mouse pointer is in.

However, it is made in Python.
Do you think I should remake it in C# (which is the only other language I'm good at besides Shell Script) so it's faster, or just continue with Python?

**Reply by redstone-dev** [reply]
I think you could rewrite it in C#.

**Reply by Nonethousand** [reply]
> I think you could rewrite it in C#.

Got it, I'll get to work soon.
Then I'll make the basics of the paint app.

**Reply by Nonethousand** [reply]
Okay, I got the thing working. Do you want me to make a repo with the code, or make the basics of the app first?

**Reply by redstone-dev** [reply]
how about make the basics of the app, then make a repo

**Reply by Nonethousand** [reply]
Got it.

**Reply by Nonethousand** [reply]
Also, can you assign me to this?

**Reply by redstone-dev** [reply]
Sure.

**Comment by Nonethousand** [comment]
Here ya go! The base of the paint app. I'll make a repo in a moment.

https://user-images.githubusercontent.com/78574005/214754110-b59a3dc2-f099-4074-8f4f-75f086e437ad.mp4

**Reply by Nonethousand** [reply]
I accidentally used a function exclusive to Windows, so I'll fix that along with a few other things

**Comment by Nonethousand** [comment]
Well, there's a problem. I used kernel32.dll for this, and I can't find a substitute for it.

The only thing I can think of now is get rid of it, or use Python instead of C#

I think I can just get rid of it, the user will just have to make sure to not resize the terminal

**Reply by redstone-dev** [reply]
ok just add a warning to not resize the terminal lololol

or you could just warn them to restart the program when they _do_ resize it

**Comment by Nonethousand** [comment]
Due to technical difficulties with my PC, I won't be able to program in C#. But I'm going to get a new one soon.
